### DISCUSSION TITLE
How does nGrinder save scripts on the gitLab server?

### DISCUSSION THREAD

**Discussion by jeff526** [naver/ngrinder#814] [Q&A]
****How does nGrinder save scripts on the gitLab server?**

**Comment by donggyu04** [comment]
@jeff526 

If GitLab API is compatible with GitHub, see this [guide](https://github.com/naver/ngrinder/wiki/GitHub-Script-Storage).
Otherwise, you can't.

**Reply by weikunlun89** [reply]
请问你的问题解决了吗？可以配置gitlab仓库吗？
我的理解是不是应该自己二次开发支持gitlab?

**Reply by junoyoon** [reply]
@weikunlun89 please USE ENGLISH.

**Reply by weikunlun89** [reply]
How does the ngrinder script support gitlab's repository？

**Reply by junoyoon** [reply]
No. we don't support it.

**Reply by weikunlun89** [reply]
Can the ngrinder script add support for gitlab repositories? Thanks

**Reply by junoyoon** [reply]
nGrinder depends on [svn protocol support](https://docs.github.com/en/github/importing-your-projects-to-github/working-with-subversion-on-github/support-for-subversion-clients) to support github. 
However as far as I know, gitlab doesn't have such a feature. Therefore. it's impossible.

You can contribute it if you really need it. You're welcome.

**Reply by weikunlun89** [reply]
Thank you for your reply. I will try my best. Ha ha, can I support Jgit in theory?

**Reply by junoyoon** [reply]
No. You should embed whole gitlab repository in nGrinder or... you should find out how to clone the git repo within few seconds.
That's why we uses svn protocol for github as well.
